### DISCUSSION TITLE
clarifications needed for fields reference in documentation

### DISCUSSION THREAD

**Discussion by plawton-umd** [NASA-PDS/pds-api#211] [General]
Hello,

Working from the https://nasa-pds.github.io/pds-api/guides/search/endpoints.html page.

The "limit=<number>" and "summary-only=true" options work with "--data-urlencode".  Thank you.

However, "fields" is returning "Method Not Allowed".

Examples, 

curl https://pds.nasa.gov/api/search-sbnumd/1.0/bundles/urn:nasa:pds:gbo-kpno::4.0/collections --data-urlencode 'fields=Time_Coordinates.start_date_time' | json_pp
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   178    0   139  100    39    134     37  0:00:01  0:00:01 --:--:--   171
{
   "error" : "Method Not Allowed",
   "message" : "",
   "path" : "/bundles/urn:nasa:pds:gbo-kpno::4.0/collections",
   "status" : 405,
   "timestamp" : 1659118809520
}

Also tried including the "pds:"

curl https://pds.nasa.gov/api/search-sbnumd/1.0/bundles/urn:nasa:pds:gbo-kpno::4.0/collections --data-urlencode 'fields=pds:Time_Coordinates.start_date_time' | json_pp
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   184    0   139  100    45    150     48 --:--:-- --:--:-- --:--:--   199
{
   "error" : "Method Not Allowed",
   "message" : "",
   "path" : "/bundles/urn:nasa:pds:gbo-kpno::4.0/collections",
   "status" : 405,
   "timestamp" : 1659118862728
}

The output of "curl https://pds.nasa.gov/api/search-sbnumd/1.0/bundles/urn:nasa:pds:gbo-kpno::4.0/collections"
includes the "pds:Time_Coordinates.pds:start_date_time" (NOTE: an extra "pds") which also has not worked in 
the examples.  Trying a simpler field name, "lid", returned the same error message.

curl https://pds.nasa.gov/api/search-sbnumd/1.0/bundles/urn:nasa:pds:gbo-kpno::4.0/collections --data-urlencode 'fields=lid' | json_pp      % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   149    0   139  100    10    276     19 --:--:-- --:--:-- --:--:--   296
{
   "error" : "Method Not Allowed",
   "message" : "",
   "path" : "/bundles/urn:nasa:pds:gbo-kpno::4.0/collections",
   "status" : 405,
   "timestamp" : 1659121356558
}

Suggestions please.

Thanks,
Pat

**Comment by jordanpadams** [comment]
@tloubrieu-jpl can you see what is going on here?

**Comment by jordanpadams** [comment]
@plawton-umd it looks like you are missing the `--get` flag:
```
curl --get https://pds.nasa.gov/api/search-sbnumd/1.0/bundles/urn:nasa:pds:gbo-kpno::4.0/collections --data-urlencode 'fields=Time_Coordinates.start_date_time' | json_pp
```

that being said, the `fields` flag looks to be not implemented/tested sufficiently across all return types. it does not appear to work with `application/json` response, and only works in some case for the `text/csv` and `application/kvp+json` response. I will create separate tickets for that.

**Comment by jordanpadams** [comment]
but for instance, it looks like this works:

```
curl --get 'https://pds.nasa.gov/api/search-sbnumd/1.0/bundles/urn:nasa:pds:gbo-kpno::4.0' --data-urlencode 'fields=pds:Time_Coordinates.pds:start_date_time' --header "Accept: text/csv"
```

**Comment by jordanpadams** [comment]
note the field is `pds:Time_Coordinates.pds:start_date_time` not `Time_Coordinates.start_date_time` will update the docs where that is wrong

**Comment by tloubrieu-jpl** [comment]
@plawton-umd @jordanpadams sorry for my late reply. Indeed --get is making the request to be an HTTP GET request which s the ne supported by the API currently. Without that option, curl makes a POST request which is not supported by the API which is why you were getting the method not allowed message.

**Comment by tloubrieu-jpl** [comment]
@jordanpadams Regarding the doc which need to be fixed it is related to sphinx postprocessing of the text which removes namespaces... I need to find a way to skip that.

**Comment by jordanpadams** [comment]
@tloubrieu-jpl hmmmm. ok we can look into that. the documentation bugs I think I actually just fixed. there were a couple instances where I found `Time_Coordinates.start_date_time`. see https://github.com/NASA-PDS/pds-api/commit/ff05a78f1d813cddc68b76df4c85feea60254c6f

**Reply by tloubrieu-jpl** [reply]
Oh, perfect, then that is because you fixed the issue when I looked at it.
